pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -45,16 +47,18 @@ func ValidateKubeSchedulerConfiguration(cc *config.KubeSchedulerConfiguration) f
 	if len(cc.Profiles) == 0 {
 		allErrs = append(allErrs, field.Required(profilesPath, ""))
 	} else {
+		var errs []error
```

Comment:
just use allErrs and flatten at the end

Reply:
How can I append a `error` to `field.ErrorList{}`? 
```golang
allErrs = append(allErrs, errs)
```
Will hit compile error
```
cannot use errs (variable of type []error) as *field.Error value in argument to append
```
`field.Error` is a `error` but not true on the other hand.

If we change `validateKubeSchedulerProfile` to return `field.ErrorList`, we still need `fromAggregate` function to flatten the `Aggregate` to `field.ErrorList`. 

Below pseudo code will hit compile issue, 
```golang
allErrs = append(allErrs, utilerrors.Aggregate)
```
```
cannot use errList (variable of type "k8s.io/apimachinery/pkg/util/errors".Aggregate) as *field.Error value in argument to append
```

Reply:
Done